src/components/drawing/index.js

```diff
@@ -290,50 +289,22 @@ drawing.pointStyle = function(s, trace) {
     // allow array marker and marker line colors to be
     // scaled by given max and min to colorscales
     var marker = trace.marker;
-    var markerIn = (trace._input || {}).marker || {},
-        markerScale = drawing.tryColorscale(marker, markerIn, ''),
-        lineScale = drawing.tryColorscale(marker, markerIn, 'line.');
+    var markerScale = drawing.tryColorscale(marker, ''),
+        lineScale = drawing.tryColorscale(marker, 'line');
 
     s.each(function(d) {
         drawing.singlePointStyle(d, d3.select(this), trace, markerScale, lineScale);
     });
 };
 
-// for a given color attribute (ie m -> mc = marker.color) look to see if we
-// have a colorscale for it (ie mscl, mcmin, mcmax) - if we do, translate
-// all numeric color values according to that scale
-drawing.tryColorscale = function(cont, contIn, prefix) {
-    var colorArray = Lib.nestedProperty(cont, prefix + 'color').get(),
-        scl = Lib.nestedProperty(cont, prefix + 'colorscale').get(),
-        auto = Lib.nestedProperty(cont, prefix + 'cauto').get(),
-        minProp = Lib.nestedProperty(cont, prefix + 'cmin'),
-        maxProp = Lib.nestedProperty(cont, prefix + 'cmax'),
-        min = minProp.get(),
-        max = maxProp.get();
-
-    // TODO handle this in Colorscale.calc
-    if(scl && Array.isArray(colorArray)) {
-        if(auto || !isNumeric(min) || !isNumeric(max)) {
-            min = Infinity;
-            max = -Infinity;
-            colorArray.forEach(function(color) {
-                if(isNumeric(color)) {
-                    if(min > color) min = +color;
```

Comment:
This block here used to override the [`Colorscale.calc`](https://github.com/plotly/plotly.js/blob/master/src/components/colorscale/calc.js) computations in the case where `marker.cauto` (or `marker.line.cauto`) was true. 

:hocho: :hocho: :hocho: 

Luckily, for most cases, this block here gave the same results as `Colorscale.calc`. One exception coming when the computated `cmin` and `cmax` where equal. In that case, `Colorscale.calc` spreads them apart by -0.5 and +0.5 respectively [here](https://github.com/plotly/plotly.js/blob/8745a22e42933b48b85ae05c7daf46bc20e7726c/src/components/colorscale/calc.js#L43-L46) whereas the removed block left them as is.